clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/model/loadbalancing/http/AbstractGoogleHttpLoadBalancer.groovy

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.google.model.loadbalancing.http
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.netflix.spinnaker.clouddriver.google.model.loadbalancing.GoogleLoadBalancer
+import groovy.transform.Canonical
+
+@Canonical
+abstract class AbstractGoogleHttpLoadBalancer extends GoogleLoadBalancer {
```

Comment:
`Abstract` seems wrong here - an abstract class generally requires a subclass to implement some method. You may want to consider `BaseGoogleHttpLoadBalancer`.

That being said, I haven't gotten through the whole PR to see what the actual need is to split the classes up.